sdk/helper/storagepacker/storagepacker_v2.go

```diff
@@ -114,6 +264,56 @@ func GetItemIDHash(itemID string) string {
 	return hex.EncodeToString(cryptoutil.Blake2b256Hash(itemID))
 }
 
+func (s *StoragePackerV2) visitDiskBucketsInOrder(ctx context.Context, keys []string) error {
+	// If we crash while writing out new buckets created by sharding,
+	// we will have version N of the bucket in storage, and version N+1
+	// represented in the sub-buckets.
+	//
+	// To bring these two into alignment, we can roll back to version N.
+	// It might be possible to foll forward in some cases (particularly if
+	// only one item can be added at a time?) but the crash must have
+	// occurred before PutBucket() returned, so the new data need not be saved.
+	//
+	// If we load a bucket that has sub-buckets but is non-empty,
+	// delete all of the sub-buckets.  (The current implementation recurses
+	// so there may be more than one level of sub-buckets created by shardBucket.)
+
+	// Visit the buckets using inorder traversal, parents before children.
+	sort.Strings(keys)
+
+	nonemptyParent := "NOT_A_PREFIX"
+
+	for _, key := range keys {
+		if strings.HasPrefix(key, nonemptyParent) {
+			// FIXME: can I give more context about the storage path?
+			s.Logger.Warn("Detected shadowed bucket, removing", "key", key, "parent", nonemptyParent)
```

Comment:
Log messages (and error messages) start lowercase

Reply:
OK, will fix.